api/src/services/graphql/utils/sanitize-gql-schema.ts

```diff
@@ -0,0 +1,101 @@
+import type { Relation, SchemaOverview } from '@directus/types';
+import { useLogger } from '../../../logger.js';
+
+/**
+ * Regex was taken from the spec
+ * https://spec.graphql.org/June2018/#sec-Names
+ */
+const GRAPHQL_NAME_REGEX = /^[_A-Za-z][_0-9A-Za-z]*$/;
+
+// Not an exhaustive list (doesnt include generated names)
```

Comment:
Do you know of a resource that we could link to?
And if it is correct that Directus-related keywords are also included, then I would mention that.

Reply:
There are no directus-related or generated keywords included as the comment says. No I dont have a link to a central place for these keywords.

Reply:
How about linking to https://graphql.org/learn/schema/ then, as most reserved names are schema types?
Where does `uid` come from, could we perhaps add a comment?
Just trying to make it a bit more tangible for me and future visitors of this code 🙏

Reply:
Since that link doesnt explicitly mention reserved keywords nor lists what they are, I dont think adding that link adds much value here tbh. These values come from various internet sources, examination of graphql-composer, and trial and error. But sure I'll update the comment to make it clearer that it's just a handpicked and tested list of reserved keywords that cannot be used with graphql 👌

Reply:
I see, sounds good ❤️